cider/type/plan/DateExpr.cpp

```diff
@@ -0,0 +1,187 @@
+    func_name = func_name + "Months";
+    switch (to_type) {
+      case daMONTH:
+        return interval;
+      case daQUARTER:
+        return interval * 3;
+      case daYEAR:
+        return interval * 12;
+      case daDECADE:
+        return interval * 120;
+      case daCENTURY:
+        return interval * 1200;
+      case daMILLENNIUM:
+        return interval * 12000;
+    }
+  }
+  CIDER_THROW(
+      CiderUnsupportedException,
+      fmt::format("date interval type is {}, field is {}", toString(from_type), to_type));
+}
+
+JITExprValue& DateaddExpr::codegen(JITFunction& func) {
+  const SQLTypeInfo& expr_ti = get_type_info();
+  CHECK(expr_ti.get_type() == kTIMESTAMP || expr_ti.get_type() == kDATE);
+  FixSizeJITExprValue datetime(get_datetime_expr()->codegen(func));
+  FixSizeJITExprValue interval(get_number_expr()->codegen(func));
+  const SQLTypeInfo& datetime_ti = get_datetime_expr()->get_type_info();
+  const SQLTypeInfo& interval_ti = get_number()->get_type_info();
+
+  std::string func_name = datetime_ti.get_type() == kDATE ? "DateAdd" : "TimeAdd";
+  JITValuePointer datetime_val = datetime.getValue();
+  JITValuePointer interval_val = getScaledIntervalAndFunc(
+      interval.getValue().get(), interval_ti.get_type(), get_field(), func_name);
+
+  if (datetime_ti.is_high_precision_timestamp()) {
+    func_name = func_name + "HighPrecision";
+    JITValuePointer dim_val =
+        func.createConstant(JITTypeTag::INT32, datetime_ti.get_dimension());
+    JITValuePointer res_val = func.emitRuntimeFunctionCall(
+        func_name,
+        JITFunctionEmitDescriptor{
+            .ret_type = JITTypeTag::INT64,
+            .params_vector = {datetime_val.get(), interval_val.get(), dim_val.get()}});
+    return set_expr_value(datetime.getNull(), res_val);
+  }
+  JITValuePointer res_val = func.emitRuntimeFunctionCall(
+      func_name,
+      JITFunctionEmitDescriptor{
+          .ret_type =
+              datetime_ti.get_type() == kDATE ? JITTypeTag::INT32 : JITTypeTag::INT64,
+          .params_vector = {datetime_val.get(), interval_val.get()}});
+  return set_expr_value(datetime.getNull(), res_val);
+}
+
+std::shared_ptr<Analyzer::Expr> DateaddExpr::rewrite_with_targetlist(
+    const std::vector<std::shared_ptr<TargetEntry>>& tlist) const {
+  return makeExpr<DateaddExpr>(type_info,
+                               field_,
+                               number_->rewrite_with_targetlist(tlist),
+                               datetime_->rewrite_with_targetlist(tlist));
+}
+
+std::shared_ptr<Analyzer::Expr> DateaddExpr::rewrite_with_child_targetlist(
+    const std::vector<std::shared_ptr<TargetEntry>>& tlist) const {
+  return makeExpr<DateaddExpr>(type_info,
+                               field_,
+                               number_->rewrite_with_child_targetlist(tlist),
+                               datetime_->rewrite_with_child_targetlist(tlist));
+}
+
+std::shared_ptr<Analyzer::Expr> DateaddExpr::rewrite_agg_to_var(
```

Comment:
For those legacy methods of `DateAddExpr`, can you add comments for the ones which we should remove in future?